mkosi/__init__.py

```diff
@@ -3717,6 +3717,9 @@ def insert_partition(
 
     luks_extra = 16 * 1024 * 1024 if args.encrypt == "all" else 0
     blob_size = os.stat(blob.name).st_size
+    if ident == PartitionIdentifier.root and not args.minimize:
+        # Make root partition at least as big as the specified size
+        blob_size = max(blob_size, args.root_size)
```

Comment:
Hmm, I think a bigger fix should be applied to this. Non only root but other partitions also may use squashfs. We should apply the same logic to them.

I know the docs for `--minimize` say "root file system", but I think that's a left-over. If we're build a usr-only system, we'd want to minimize it too.

I think we need to first figure out what the desired user-facing logic is…

Reply:
If `UsrOnly` is set then this logic will apply to that partition type too. Based on the `PartitionIdentifier` Enum class we have the following partition types:

  * `root` or `usr` depending on the `UsrOnly` option. The code however treats them both as `PartitionIdentifier.root`.
  * `verity` and `verity-sig` partition types which are always automatically sized. It would be ideal to also have size controls for these too but that is another PR for another day.
  * An `esp` partition type which has a default size and an ability to set an explicit size
  * Other partition types under `PartitionIdentifier` which each have an explicit control for specifying their individual sizes

I would suggest the following logic as below.

| Partition Type | SquashFS | Minimize | Outcome | Matches Current Behaviour |
|--------------------|---------------|--------------|--------------|--------------------------------------|
| `root` or `usr` |  False        | False       | Use explicit root size with fallback to default | True |
| `root` or `usr` |  False        | True        | Use the minimized size for the partition even if the root size was set | True |
| `root` or `usr` |  True        | False        | Use explicit root size with fallback to minimized size | False |
| `root` or `usr` |  True        | True          | Use the minimized size for the partition even if the root size was set | True |
| `esp` | N/A | N/A | Use explicit size with fallback to default | True |
| `verity` and `verity-sig` | N/A | N/A | Use minimized size | True |
| All others | N/A | N/A | Use explicit size | True |

Reply:
@keszybz Did you get a chance to look over the feedback I provided?